src/static/js/ace2_inner.js

```diff
@@ -2471,12 +2474,14 @@ function Ace2Inner(){
 
     if (selectionAllHasIt)
     {
+      ifAttribIsFormattingUpdateStyleButtonState(attributeName, false); // italic, bold, ...
       documentAttributeManager.setAttributesOnRange(rep.selStart, rep.selEnd, [
         [attributeName, '']
       ]);
     }
     else
     {
+      ifAttribIsFormattingUpdateStyleButtonState(attributeName, true); // italic, bold, ...
```

Comment:
these two calls of `ifAttribIsFormattingUpdateStyleButtonState` can be simplified into a single one, passing `! selectionAllHasIt` as the second parameter:
```
    if (selectionAllHasIt)
    {
      documentAttributeManager.setAttributesOnRange(rep.selStart, rep.selEnd, [
        [attributeName, '']
      ]);
    }
    else
    {
      documentAttributeManager.setAttributesOnRange(rep.selStart, rep.selEnd, [
        [attributeName, 'true']
      ]);
    }
    ifAttribIsFormattingUpdateStyleButtonState(attributeName, !selectionAllHasIt); // italic, bold, ...
```

Reply:
BTW, it is even possible to simplify the code even more:
```
var attributeValue = selectionAllHasIt ? '' : 'true';
documentAttributeManager.setAttributesOnRange(rep.selStart, rep.selEnd, [[attributeName, attributeValue]]);
ifAttribIsFormattingUpdateStyleButtonState(attributeName, !selectionAllHasIt); // italic, bold, ...
```